### PR TITLE
Feature/dtserwthre 299 update transfer method info

### PR DESCRIPTION
**Pull request by ddhar-hw**
Has the end to end functionality for update transfer method. 
Jira tickets involved :

- https://engineering.paypalcorp.com/jira/browse/DTSERWTHRE-298
- https://engineering.paypalcorp.com/jira/browse/DTSERWTHRE-279
- https://engineering.paypalcorp.com/jira/browse/DTSERWTHRE-299

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,182 @@
+GEM
```
Comment (jsaini-hw) [inline Gemfile.lock:1]: Why was this file added?

File: HyperwalletUISDK.xcodeproj/project.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jsaini-hw) [inline HyperwalletUISDK.xcodeproj/project.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings:1]: revert the changes

File: Makefile [hunk 3]
```diff
@@ -9,6 +9,7 @@ docs:
 		sourcekitten doc --module-name ReceiptRepository > TempJson/ReceiptRepository.json
 		sourcekitten doc --module-name Receipt > TempJson/Receipt.json
 		@jazzy \
+					--build-tool-arguments CODE_SIGN_IDENTITY=,CODE_SIGNING_REQUIRED=NO,CODE_SIGNING_ALLOWED=NO
```
Comment (jsaini-hw) [inline Makefile:12]: how are these changes in your ticket? Please revert them.

File: TransferMethod/Sources/View/Widget/AbstractWidget.swift [hunk 4]
```diff
@@ -127,9 +128,27 @@ class AbstractWidget: UIStackView, UITextFieldDelegate {
...
+                                                              to: textField.endOfDocument)
+        }
+    }
+
+    func textField(_ textField: UITextField,
+                   shouldChangeCharactersIn range: NSRange,
+                   replacementString string: String) -> Bool {
+        if let isValueMasked = field.fieldValueMasked, isValueMasked && !isValueUpdated {
+            textField.text = ""
+            isValueUpdated = true
```
Comment (jsaini-hw) [inline TransferMethod/Sources/View/Widget/AbstractWidget.swift:142]: This should be outside of the if `isValueMasked = field.fieldValueMasked, isValueMasked ` block. isValueUpdated should be set to true for any other field as well that is being updated but should be set only once

File: TransferMethod/Sources/View/Widget/AbstractWidget.swift [hunk 5]
```diff
@@ -127,9 +128,27 @@ class AbstractWidget: UIStackView, UITextFieldDelegate {
...
+        }
+        return true
     }
 
     func textFieldDidEndEditing(_ textField: UITextField) {
+        if let isValueMasked = field.fieldValueMasked, isValueMasked && !isValueUpdated {
+            return
+        }
+        isValueUpdated = true
```
Comment (jsaini-hw) [inline TransferMethod/Sources/View/Widget/AbstractWidget.swift:151]: This is not required here. It should be set to true in `func textField(_ textField: UITextField,
                   shouldChangeCharactersIn range: NSRange,
                   replacementString string: String) -> Bool`  for all other fields as well

File: Gemfile.lock (hunk 1 above)
Comment (ddhar-hw) [reply Gemfile.lock]: My bad, I added this by mistake while merging branches

**Review by jsaini-hw — APPROVED** [review]

**Review by nanbehera — APPROVED** [review]
